### PR TITLE
Add glossary for typically misused terms

### PR DESCRIPTION
**Pull request by z1c0**
This PR introduces a glossary for OpenTelemetry specific terms that have caused confusion and ambiguity issues over time.
The glossary is not exhaustive and intended to get updated over time.

### CODE REVIEW

**Comment by carlosalberto** [comment]
Looks good to me - I'd like only to verify the namespace aspect related to metrics.

File: specification/glossary.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+Denotes the actual *OpenTelemetry* (or API-compatible) library in use.
+
+See [Library resource semantic conventions](data-resource-semantic-conventions.md#library)
+
+<a name="instrumented_library"></a>
+
+## Instrumented Library
+
+Denotes the library that is monitored, i.e. the library of interest that signals (traces, metrics, logs) are emitted for.
```
Comment (yurishkuro) [inline specification/glossary.md]:
```suggestion
Denotes the library for which the telemetry signals (traces, metrics, logs) are gathered.
```

File: specification/glossary.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+Example: `org.mongodb.client`.
+
+<a name="instrumenting_library"></a>
+
+## Instrumenting Library
+
+Denotes the library which provides the instrumentation *for* a given library
+(i.e. *for* the [Instrumented Library](#instrumented_library)). Hence,
+*Instrumented Library* and *Instrumenting Library* can be identical if a library
+has built-in OpenTelemetry instrumentation, but they don't necessarily have to.
```
Comment (yurishkuro) [inline specification/glossary.md]:
```suggestion
*Instrumented Library* and *Instrumenting Library* may be the same library
if it has built-in OpenTelemetry instrumentation.
```

File: specification/glossary.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+happen in a different one (the [Instrumenting Library](#instrumenting_library)).
+
+Example: `org.mongodb.client`.
+
+<a name="instrumenting_library"></a>
+
+## Instrumenting Library
+
+Denotes the library which provides the instrumentation *for* a given library
+(i.e. *for* the [Instrumented Library](#instrumented_library)). Hence,
```
Comment (yurishkuro) [inline specification/glossary.md]:
```suggestion
Denotes the library that provides the instrumentation for a given [Instrumented Library](#instrumented_library). 
```

File: specification/glossary.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+See [Library resource semantic conventions](data-resource-semantic-conventions.md#library)
+
+<a name="instrumented_library"></a>
+
+## Instrumented Library
+
+Denotes the library that is monitored, i.e. the library of interest that signals (traces, metrics, logs) are emitted for.
+
+The actual OpenTelemetry API calls can happen inside this library but might as well
+happen in a different one (the [Instrumenting Library](#instrumenting_library)).
```
Comment (yurishkuro) [inline specification/glossary.md]:
```suggestion
The calls to the OpenTelemetry API can be done either by the Instrumented Library itself, 
or by another [Instrumenting Library](#instrumenting_library).
```

File: specification/glossary.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
+# Glossary
+
+Below are a list of some OpenTelemetry specific terms that are used across this
+specification.
+
+## (Telemetry) Library
+
+Denotes the actual *OpenTelemetry* (or API-compatible) library in use.
```
Comment (yurishkuro) [inline specification/glossary.md]:
```suggestion
Denotes the library that implements the *OpenTelemetry API*.
```

File: specification/glossary.md (hunk 1 above)
Comment (Oberon00) [reply specification/glossary.md]: I liked the old version better

Comment (carlosalberto) [reply specification/glossary.md]: I think both versions are fine, so I'd leave it to to the author to decide this one ;)

Comment (yurishkuro) [reply specification/glossary.md]: The old version:
  * poor English (ends sentence with a preposition)
  * uses inaccurate word "emitted". Telemetry may be gathered but never emitted.

File: specification/glossary.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
+# Glossary
+
+Below are a list of some OpenTelemetry specific terms that are used across this
+specification.
+
+## (Telemetry) Library
+
+Denotes the library that implements the *OpenTelemetry API*.
+
+See [Library resource semantic conventions](data-resource-semantic-conventions.md#library)
```
Comment (arminru) [inline specification/glossary.md]: from https://github.com/open-telemetry/opentelemetry-specification/pull/494#discussion_r387142820:
```suggestion
## Telemetry SDK

Denotes the library that implements the *OpenTelemetry API*.

See [Library Guidelines](library-guidelines.md#sdk-implementation) and
[Library resource semantic conventions](data-resource-semantic-conventions.md#telemetry-sdk)
```

File: specification/glossary.md (hunk 1 above)
Comment (z1c0) [reply specification/glossary.md]: Fair enough @yurishkuro; I already applied your suggestion.

**Pull request merged**